### PR TITLE
support for zero node clusters in swarm container service

### PR DESCRIPTION
**Pull request by arafato**
See https://partners-intl.aliyun.com/help/doc-detail/25995.htm for details on zero-node cluster.

Unit-Test Output:
=== RUN   TestAccAlicloudCSSwarm_vpc
--- PASS: TestAccAlicloudCSSwarm_vpc (575.64s)
=== RUN   TestAccAlicloudCSSwarm_vpc_zero_node
--- PASS: TestAccAlicloudCSSwarm_vpc_zero_node (326.74s)
=== RUN   TestAccAlicloudCSSwarm_vpc_noslb
--- PASS: TestAccAlicloudCSSwarm_vpc_noslb (571.62s)
PASS
ok  	github.com/terraform-providers/terraform-provider-alicloud/alicloud	1474.059s

### CODE REVIEW

File: alicloud/resource_alicloud_cs_swarm_test.go [hunk 1]
```diff
@@ -101,6 +101,31 @@ func TestAccAlicloudCSSwarm_vpc(t *testing.T) {
...
+
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckSwarmClusterDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccCSSwarm_basic_zero_node,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckContainerClusterExists("alicloud_cs_swarm.cs_vpc", &container),
+					resource.TestCheckResourceAttr("alicloud_cs_swarm.cs_vpc", "node_number", "0"),
+					resource.TestCheckResourceAttr("alicloud_cs_swarm.cs_vpc", "nodes.#", "0"),
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_cs_swarm_test.go]: Please check all attributes?

In addition, can you add another step to test updating node number from 0 to 2?

**Comment by arafato** [comment]
done. also added new unit test for update operation:
=== RUN   TestAccAlicloudCSSwarm_vpc_zero_node_update
--- PASS: TestAccAlicloudCSSwarm_vpc_zero_node_update (676.18s)
PASS
ok  	github.com/terraform-providers/terraform-provider-alicloud/alicloud	676.241s

File: alicloud/resource_alicloud_cs_swarm.go [hunk 2]
```diff
@@ -259,7 +263,11 @@ func resourceAlicloudCSSwarmUpdate(d *schema.ResourceData, meta interface{}) err
 		}
 
 		_, err = client.WithCsClient(func(csClient *cs.Client) (interface{}, error) {
-			return nil, csClient.WaitForClusterAsyn(d.Id(), cs.Running, 500)
+			state := cs.Running
+			if int64(ni) == 0 {
```
[the remark this answers is not in the document]
Comment (arafato) [reply alicloud/resource_alicloud_cs_swarm.go]: done

File: alicloud/resource_alicloud_cs_swarm_test.go (hunk 1 above)
Comment (arafato) [reply alicloud/resource_alicloud_cs_swarm_test.go]: done.

Comment (xiaozhu36) [reply alicloud/resource_alicloud_cs_swarm_test.go]: yes, I know. So, it should check disk size is 0 and disk type is "".

In addition, can you add another step to test updating node number from 0 to 2?

Comment (arafato) [reply alicloud/resource_alicloud_cs_swarm_test.go]: done. also already added additional unit tests (called "TestAccAlicloudCSSwarm_vpc_zero_node_update") to check updating node number from 0 to 2:
=== RUN   TestAccAlicloudCSSwarm_vpc_zero_node
--- PASS: TestAccAlicloudCSSwarm_vpc_zero_node (357.06s)
=== RUN   TestAccAlicloudCSSwarm_vpc_zero_node_update
--- PASS: TestAccAlicloudCSSwarm_vpc_zero_node_update (693.68s)
PASS
ok	github.com/terraform-providers/terraform-provider-alicloud/alicloud	1050.74

**Comment by arafato** [comment]
Updated unit tests and also DiffSupressFunc:
=== RUN   TestAccAlicloudCSSwarm_vpc_zero_node
--- PASS: TestAccAlicloudCSSwarm_vpc_zero_node (329.54s)
=== RUN   TestAccAlicloudCSSwarm_vpc_zero_node_update
--- PASS: TestAccAlicloudCSSwarm_vpc_zero_node_update (876.45s)
PASS
ok  	github.com/terraform-providers/terraform-provider-alicloud/alicloud	1206.064s

**Pull request merged**